src/odemis/driver/piezomotor.py

```diff
@@ -425,26 +425,32 @@ def _waitEndMotion(self, f, shifts):
 
         end_time = time.time() + max_dur
         moving_axes = set(shifts.keys())  # All axes (still) moving
+        logging.debug(f"Axes {moving_axes} are moving.")
         while moving_axes:
             if f._must_stop.is_set():
                 for axname in moving_axes:
                     self.stopAxis(self._axis_map[axname])
                     raise CancelledError()
 
             if time.time() > end_time:
-                raise TimeoutError("Timeout after while waiting for end of motion on axes %s for %g s" % (moving_axes, max_dur))
+                raise TimeoutError(
+                    "Timeout while waiting for end of motion on axes %s for %g s" % (moving_axes, max_dur))
```

Comment:
Why don't you also use f strings in the error message?

Reply:
I just reformatted this error so it would fit the line length. I used the f strings for new errors and log messages.